src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -383,6 +383,11 @@ private static bool IsProcessInList(string processName, string[] processList)
         private static string GetProcessFromRunningProcess()
         {
 #if RUNTIME_TYPE_NETCORE
+            if (AssemblyUtilities.EntryAssembly == null)
```

Comment:
Feels to me like this should be pushed down into AssemblyUtilities instead. Maybe add a new method there that encapsulates all this?

@rainersigwald what do you think?

Reply:
A property like `AssemblyUtilities.EntryAssemblyFilePath` can encapsulate this logic, but I don't think that it is a good idea, because in my vision core classes like `AssemblyUtilities` must expose a real state of the environment.